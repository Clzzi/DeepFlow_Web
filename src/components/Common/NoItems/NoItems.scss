.NoItems {
  display: flex;
  display: -webkit-flex;
  flex-flow: column wrap;
  align-items: center;
  margin-top: 3rem;

  &-Image {
    width: 50%;
  }

  &-Text {
    text-align: center;
    font-size: var(--font-contents);
    color: var(--contrast);
    margin-top: 1.5rem;
  }
}
