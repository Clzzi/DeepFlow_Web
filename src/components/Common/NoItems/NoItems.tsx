import { memo } from 'react';
import classNames from 'classnames';
import { ClassNamesFn } from 'classnames/types';
import { ReactComponent as NoPost } from 'assets/images/no-post.svg';

const style = require('./NoItems.scss');
const cx: ClassNamesFn = classNames.bind(style);

interface NoItemsProps {
  text: string;
  imageWidth?: string;
}

const NoItems = ({
  text,
  imageWidth = '45%',
}: NoItemsProps): JSX.Element => {
  return (
    <div className={cx('NoItems')}>
      <NoPost
        className={cx('NoItems-Image')}
        style={{ width: imageWidth }}
      />
      <div className={cx('NoItems-Text')}>{text}</div>
    </div>
  );
};

export default memo(NoItems);
