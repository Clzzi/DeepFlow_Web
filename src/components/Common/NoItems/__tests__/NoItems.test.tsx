import { render } from '@testing-library/react';
import NoItems, { NoItemsProps } from '../NoItems';

describe('NoItems', () => {
  const initialSetup = (props: Partial<NoItemsProps> = {}) => {
    const initialProps: NoItemsProps = {
      text: '',
      imageWidth: '45%',
      ...props,
    };

    return render(<NoItems {...initialProps} />);
  };

  it('render correctly', () => {
    const { container } = initialSetup();
    expect(container).toMatchSnapshot();
  });

  it('props render correctly', () => {
    const wrapper = initialSetup({ text: '목록이 없습니다.', imageWidth: '45%' });
    expect(wrapper.getByTestId('no-items-text')).toMatchSnapshot();
    expect(wrapper.getByTestId('no-items-image')).toMatchSnapshot();
  });
});