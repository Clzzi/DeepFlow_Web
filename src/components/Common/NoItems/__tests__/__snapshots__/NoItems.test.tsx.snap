// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoItems props render correctly 1`] = `
<div
  class="NoItems-Text"
  data-testid="no-items-text"
>
  목록이 없습니다.
</div>
`;

exports[`NoItems props render correctly 2`] = `
<img
  alt="noPost"
  class="NoItems-Image"
  data-testid="no-items-image"
  src="no-post.svg"
  style="width: 45%;"
/>
`;

exports[`NoItems render correctly 1`] = `
<div>
  <div
    class="NoItems"
  >
    <img
      alt="noPost"
      class="NoItems-Image"
      data-testid="no-items-image"
      src="no-post.svg"
      style="width: 45%;"
    />
    <div
      class="NoItems-Text"
      data-testid="no-items-text"
    />
  </div>
</div>
`;
